src/Symfony/Component/Security/Core/Util/ClassUtils.php

```diff
@@ -54,12 +40,6 @@ private function __construct()
      */
     public static function getRealClass($object)
     {
-        $class = is_object($object) ? get_class($object) : $object;
-
-        if (false === $pos = strrpos($class, '\\'.self::MARKER.'\\')) {
-            return $class;
-        }
-
-        return substr($class, $pos + self::MARKER_LENGTH + 2);
+        return AclClassUtils::getRealClass($object);
```

Comment:
you need to check whether the class exists first, because it is in a different pacjage which is not a dependency of ``symfony/security-core``.

and btw, the only way to be BC is to avoid depending on ``symfony/security-acl`` for the BC layer